src/state/stake/useFarmsPortfolio.ts

```diff
@@ -0,0 +1,212 @@
+
+  const complexRewardsContractAdressList =
+    complexRewarderContractList?.map(contract => (complexRewarderContractList ? contract?.address : undefined)) ?? []
+
+  const pendingComplexRewards = useMultipleContractSingleData(
+    complexRewardsContractAdressList,
+    new Interface(COMPLEX_REWARDER),
+    'pendingTokens',
+    [0, account, '0']
+  )
+
+  const complexRewardsLoading = callResultIsLoading(pendingComplexRewards)
+
+  const mapDoubleRewardToken = (addressToSearch: string) => {
+    const farm = v2Farms.find(farm => farm.doubleRewardToken.address === addressToSearch)
+    return farm?.doubleRewardToken
+  }
+
+  const earnedComplexRewardPool =
+    pendingComplexRewards.length && !complexRewardsLoading
+      ? pendingComplexRewards.map(farmData => ({
+          rewardAmount: JSBI.BigInt(farmData?.result?.rewardAmounts?.[0] ?? 0),
+          rewardToken: mapDoubleRewardToken(farmData?.result?.rewardTokens[0]) ?? dummyToken
+        }))
+      : null
+
+  const complexTokenAmounts = earnedComplexRewardPool?.map(pool => ({
+    token: pool.rewardToken,
+    tokenAddr: pool.rewardToken.address,
+    amount: new TokenAmount(AURORA[ChainId.AURORA], JSBI.BigInt(pool.rewardAmount))
+  }))
+
+  const complexRewardsFarmAmounts: FarmAmount = {}
+
+  complexTokenAmounts?.forEach(tokenReward => {
+    const tokenSymbol = tokenReward.token?.symbol ?? ''
+    return complexRewardsFarmAmounts.hasOwnProperty(tokenSymbol)
+      ? (complexRewardsFarmAmounts[tokenSymbol].amount = complexRewardsFarmAmounts[tokenSymbol].amount.add(
+          tokenReward.amount
+        ))
+      : (complexRewardsFarmAmounts[tokenSymbol] = {
+          token: tokenReward.token,
+          amount: tokenReward.amount,
+          address: tokenReward.tokenAddr
+        })
+  })
+
+  const complexRewardsFriendlyFarmAmounts = Object.entries(complexRewardsFarmAmounts).map(([name, value]) => ({
+    token: value.token,
+    tokenSymbol: name,
+    amount: value.amount.toFixed(6),
+    address: value.address
+  }))
+
+  //
+  //
+  // Tri rewards
+
+  const pendingTriDataV1 = useSingleContractMultipleData(
+    farmsReady ? contractv1 : null,
+    'pendingTri',
+    farmsReady ? v1Farms.map(farm => farm.v1args) : []
+  )
+
+  const pendingTriDataV2 = useSingleContractMultipleData(
+    farmsReady ? contractv2 : null,
+    'pendingTri',
+    farmsReady ? v2Farms.map(farm => farm.v1args) : []
+  )
+
+  const allPendingTri = [...pendingTriDataV1, ...pendingTriDataV2]
+
+  const pendingTriLoading = callResultIsLoading(allPendingTri)
+
+  const earnedTriRewardPools = allPendingTri.length
+    ? allPendingTri.map(farmData => JSBI.BigInt(farmData?.result?.[0] ?? 0))
+    : []
+
+  const allEarnedTriAmounts = earnedTriRewardPools.length
+    ? earnedTriRewardPools
+        .map(poolReward => new TokenAmount(TRI[ChainId.AURORA], JSBI.BigInt(poolReward)))
+        .reduce((a: TokenAmount, b) => a.add(b))
```

Comment:
Does this need an initializer?

```suggestion
        .reduce((a: TokenAmount, b) => a.add(b), new TokenAmount(TRI[ChainId.AURORA], BIG_INT_ZERO))
```

Reply:
It fails if I add that modification, I assume since the adding tokens are different:

<img width="546" alt="Screen Shot 2022-02-05 at 18 21 14" src="https://user-images.githubusercontent.com/96993065/152659459-9578a5a3-0fce-4af8-a003-f092e84a1adb.png">

it works if I initialize this way, with a TRI token.

```
  const allEarnedTriAmounts = earnedTriRewardPools.length
    ? earnedTriRewardPools
        .map(poolReward => new TokenAmount(TRI[ChainId.AURORA], JSBI.BigInt(poolReward)))
        .reduce((a: TokenAmount, b) => a.add(b), new TokenAmount(TRI[ChainId.AURORA], BIG_INT_ZERO))
```

===

It works with the current PR code, but I added the initialization anyway.